sdk/trace/flow_span_processor.go

```diff
@@ -0,0 +1,171 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace // import "go.opentelemetry.io/otel/sdk/trace"
+
+// Copied from https://github.com/MrAlias/flow for demo purposes.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/internal/global"
+)
+
+const (
+	startedState = "started"
+	endedState   = "ended"
+
+	// DefaultListenPort is the port the HTTP server listens on if not
+	// configured with the WithListenAddress option.
+	DefaultListenPort = 41820
+	// DefaultListenAddress is the listen address of the HTTP server if not
+	// configured with the WithListenAddress option.
+	DefaultListenAddress = ":41820"
+)
+
+type spanProcessor struct {
+	wrapped SpanExporter
+
+	idleConnsClosed  chan struct{}
+	server           *http.Server
+	spanCounter      *prometheus.CounterVec
+	exportErrCounter *prometheus.CounterVec
+}
+
+// Wrap returns a wrapped version of the downstream SpanExporter with
+// telemetry flow reporting. All calls to the returned SpanProcessor will
+// introspected for telemetry data and then forwarded to downstream.
+func Wrap(downstream SpanExporter, options ...Option) SpanProcessor {
```

Comment:
Would you want to return a `SpanExporter` here?  That way you can still use the batch processor, and you would just be instrumenting the calls to around `ExportSpans()`

Reply:
Yeah, that could makes sense. This was (quickly) adapted from the [flow](https://github.com/MrAlias/flow) `SpanProcessor`, which exports metrics about started/stopped spans. I can definitely see another package where this is a simple wrapper of a `SpanExporter` and only reports dropped values.